symfit/core/minimizers.py

```diff
@@ -64,8 +67,22 @@ class GradientMinimizer(BaseMinimizer):
     def __init__(self, *args, **kwargs):
         jacobian = kwargs.pop('jacobian')
         super(GradientMinimizer, self).__init__(*args, **kwargs)
-        self.jacobian = jacobian
-
+        def _jacobian(*args, **kwargs):
+            # Assume the order of what jacobian returns and parameters is the same
+            # We need to remove the values for the fixed parameters from what
+            # jacobian returns
+            fixed_vals = {p.name: p.value for p in self._fixed_params}
+            kwargs.update(fixed_vals)
+            out = jacobian(*args, **kwargs)
+            jac = []
+            for param, val in zip(self.parameters, out):
+                if not param.fixed:
```

Comment:
since you forced the jacobian to be a numpy array (why?) it's cleaner to use numpy masking:
```python
mask = [p.fixed for p in self.parameters]
jac = out[mask]
```

Reply:
Because it's not definitely a np array here. The scipy wrapper makes it's objective and jacobian return numpy arrays, to appease the minimizer gods.